### DISCUSSION TITLE
Weird Vram issue

### DISCUSSION THREAD

**Discussion by DivanoDova** [lshqqytiger/stable-diffusion-webui-directml#49] [Q&A]
Sometime my GPU doesn't allocate any VRAM and keep being stuck at 0mhz, so any image take minutes to generate, for fixing this issue I need to go to AMD adrenaline and change my GPU settings to Default to VRAM Overclocked to make it work again for some time then the issue reappears and I need to do this again...

Does this happen to someone else?

CPU: AMD 5600G
RAM: 32GB
GPU: AMD Rx 6600 8gb
COMMANDLINE_ARGS=--autolaunch --use-cpu interrogate --api --precision full --no-half --no-half-vae --opt-split-attention --opt-sub-quad-attention --medvram --always-batch-cond-uncond

**Comment by qwerkilo** [comment]
set COMMANDLINE_ARGS=--opt-split-attention --medvram --no-half --precision full --no-half-vae --use-cpu interrogate gfpgan codeformer --always-batch-cond-uncond --disable-nan-check --autolaunch

**Reply by qwerkilo** [reply]
or
set HSA_OVERRIDE_GFX_VERSION=10.3.0
set LD_PRELOAD=libtcmalloc.so.4
set PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.6,max_split_size_mb:128
set COMMANDLINE_ARGS=--opt-sub-quad-attention --no-half --precision full --no-half-vae --sub-quad-q-chunk-size 512 --sub-quad-kv-chunk-size 512 --sub-quad-chunk-threshold 80 --use-cpu interrogate gfpgan codeformer --disable-nan-check --autolaunch

**Reply by Miraihi** [reply]
Do `PYTORCH_CUDA` arguments really work on directml?

**Reply by Aptronymist** [reply]
> Do `PYTORCH_CUDA` arguments really work on directml?

For what it's worth, ChatGPT seemed to think that PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.6,max_split_size_mb:128 did at least (even without using cuda), though I'm testing out some that are specifically for DirectML (unsure if they do anything at all).

EDIT: Wait, I take that back, it was apparently confused and gave me conflicting answers when asking in two separate chats. I'm assuming it doesn't do squat, but I'll personally test it at a few settings to make sure.

set PYTORCH_DML_WORKSPACE_LIMIT_MB=4096
set PYTORCH_DML_WEIGHT_CACHE_CAP_MB=512
set PYTORCH_DML_ENABLE_CONCURRENT_EXECUTION=1
set PYTORCH_DML_AUTOTUNE_LEVEL=2
set PYTORCH_DML_CACHE_PATH= (doesn't seem to do anything, no files are created in path)

If these aren't doing anything at present, perhaps there is something that can be done to ensure that they start doing something...? They might contribute a fair amount to performance increases.

**Comment by Aptronymist** [comment]
> Sometime my GPU doesn't allocate any VRAM and keep being stuck at 0mhz, so any image take minutes to generate, for fixing this issue I need to go to AMD adrenaline and change my GPU settings to Default to VRAM Overclocked to make it work again for some time then the issue reappears and I need to do this again...
> 
> Does this happen to someone else?
> 
> CPU: AMD 5600G RAM: 32GB GPU: AMD Rx 6600 8gb COMMANDLINE_ARGS=--autolaunch --use-cpu interrogate --api --precision full --no-half --no-half-vae --opt-split-attention --opt-sub-quad-attention --medvram --always-batch-cond-uncond

I'm not certain on this by any stretch of the imagination, but I don't think using more than one --opt at a time does anything. I've always seen the best results from sub-quad, at least for avoiding OOM, and you should probably set something like this:
--opt-sub-quad-attention --sub-quad-q-chunk-size 256 --sub-quad-kv-chunk-size 256 --sub-quad-chunk-threshold 70 --disable-nan-check

(or 512 rather than 256) rather than the default options for opt-sub-quad-attention, which are "1024, 0, 0" iirc.
